apps/api/src/app/workflows-v2/usecases/patch-workflow/patch-workflow.usecase.ts

```diff
@@ -0,0 +1,76 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { UserSessionData, WorkflowResponseDto } from '@novu/shared';
+import { NotificationTemplateEntity, NotificationTemplateRepository } from '@novu/dal';
+import { GetWorkflowByIdsUseCase } from '@novu/application-generic';
+import { PostProcessWorkflowUpdate } from '../post-process-workflow-update';
+import { PatchWorkflowCommand } from './patch-workflow.command';
+import { GetWorkflowUseCase } from '../get-workflow';
+
+@Injectable()
+export class PatchWorkflowUsecase {
+  constructor(
+    private getWorkflowByIdsUseCase: GetWorkflowByIdsUseCase,
+    private notificationTemplateRepository: NotificationTemplateRepository,
+    private postProcessWorkflowUpdate: PostProcessWorkflowUpdate,
+    private getWorkflowUseCase: GetWorkflowUseCase
+  ) {}
+
+  async execute(command: PatchWorkflowCommand): Promise<WorkflowResponseDto> {
+    const persistedWorkflow = await this.fetchWorkflow(command);
+    let transientWorkflow = this.patchWorkflowFields(persistedWorkflow, command);
+
+    transientWorkflow = await this.postProcessWorkflowUpdate.execute({
+      workflow: transientWorkflow,
+      user: command.user,
+    });
+    await this.persistWorkflow(transientWorkflow, command.user);
+
+    return await this.getWorkflowUseCase.execute({
+      identifierOrInternalId: command.identifierOrInternalId,
+      user: command.user,
+    });
+  }
+
+  private patchWorkflowFields(persistedWorkflow, command: PatchWorkflowCommand) {
+    const transientWorkflow = { ...persistedWorkflow };
```

Comment:
What about the other workflow fields, such as name, description, and tags? Since we got to the point and of all them belong to the same form, we should be able to patch all of them at once and ease the FE.

Reply:
true, but they are also accessible from upsert, this one was a blocker for @BiswaViraj , the structure is there, we can add what ever we want with relatively small effort, was not trying to solve all the problems, only the one in front of me in an extendable way